editor/app/src/main/java/ca/ubc/eml/soiltopargraphy/editor/ui/terrain/TerrainListFragment.java

```diff
@@ -0,0 +1,118 @@
+        mRecyclerView.setLayoutManager(mLayoutManager);
+
+        TerrainAdapter terrainAdapter = new TerrainAdapter();
+        mRecyclerView.setAdapter(terrainAdapter);
+
+        Button addButton = terrainListView.findViewById(R.id.addButton);
+
+        addButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                FragmentManager manager = getActivity().getSupportFragmentManager();
+                FragmentTransaction transaction = manager.beginTransaction();
+                Fragment newFragment = new AddTerrainFragment();
+                transaction.replace(R.id.container, newFragment);
+                transaction.commit();
+            }
+        });
+
+        return terrainListView;
+    }
+
+    @Override
+    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
+        super.onActivityCreated(savedInstanceState);
+        terrainViewModel = ViewModelProviders.of(this).get(TerrainListViewModel.class);
+
+        TerrainAdapter terrainAdapter = new TerrainAdapter();
+
+        // Submits a new updated list of terrains to the terrain adapter when a change is observed
+        // in the terrains table in the room database via LiveData
+        terrainViewModel.getTerrains().observe(this, list -> terrainAdapter.submitList(list));
+    }
+
+}
+
+class TerrainAdapter extends ListAdapter<Terrain, TerrainViewHolder> {
+
+    public TerrainAdapter() {
+        super(TerrainAdapter.DIFF_CALLBACK);
+    }
+
+    @Override
+    public TerrainViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        View itemView = LayoutInflater.from(parent.getContext())
+                .inflate(R.layout.terrain_list_fragment_row, parent, false);
+
+        return new TerrainViewHolder(itemView);
+    }
+
+    @Override
+    public void onBindViewHolder(TerrainViewHolder holder, int position) {
+        holder.bindTo(this.getItem(position));
+    }
+
+    //Calculates updates for the list of terrains to be displayed
+    public static final DiffUtil.ItemCallback<Terrain> DIFF_CALLBACK =
+            new DiffUtil.ItemCallback<Terrain>() {
+                @Override
+                public boolean areItemsTheSame(@NonNull Terrain oldTerrain, @NonNull Terrain newTerrain) {
+                    return oldTerrain.getTerrainId() == newTerrain.getTerrainId();
+                }
+                @Override
+                public boolean areContentsTheSame(@NonNull Terrain oldTerrain, @NonNull Terrain newTerrain) {
+                    return oldTerrain.equals(newTerrain);
+                }
+            };
+
+}
+
+class TerrainViewHolder extends RecyclerView.ViewHolder {
+    private TextView terrainId;
+
+    public TerrainViewHolder(View view) {
+        super(view);
+        terrainId = view.findViewById(R.id.nameTextView);
+    }
+
+    public void bindTo(Terrain terrain) {
+        this.terrainId.setText(terrain.getTerrainId());
```

Comment:
nitpick: `this` isn't needed
```suggestion
        terrainId.setText(terrain.getTerrainId());
```